src/snowflake/connector/config_parser/parser.py

```diff
@@ -0,0 +1,188 @@
+#
+# Copyright (c) 2012-2023 Snowflake Computing Inc. All rights reserved.
+#
+
+from __future__ import annotations
+
+import logging
+import os
+from collections.abc import Iterable
+from functools import wraps
+from pathlib import Path
+from typing import Callable, Literal, TypeVar
+
+import tomlkit
+from tomlkit import TOMLDocument
+from tomlkit.items import Table
+
+from ..errors import ConfigParserError, ConfigSourceError
+
+_T = TypeVar("_T")
+
+LOGGER = logging.getLogger(__name__)
+
+
+class ConfigOption:
+    def __init__(
+        self,
+        name: str,
+        _root_parser: ConfigParser,
+        _nest_path: list[str],
+        _type: Callable[[str], _T] | None = None,
+        choices: Iterable[_T] | None = None,
```

Comment:
what is `choices` here? the `Iterable[_T]` type hint makes this a little confusing to me. Do we only expect a fixed/constant set of choices -- I'm thinking of `Enum` for the `choices`

Reply:
It'd be not an enum, but a list (or tuple) of possible values.